### PR TITLE
Gql05 user delete by id ap itests mammadli

### PR DESCRIPTION
**Pull request by itmelia**

### CODE REVIEW

File: src/modules/user/_test/usersGetAllMam.spec.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                variables: usersGetAllArg
+            }
+
+            requestGql(postData)
+                .expect(200)
+                .end((err, res) => {
+                    if(err) return done(err);
+                    const respData = res.body.data
+
+                    expect(respData.usersGetAll.length).not.to.eq(0)
```
Comment (rustamkutovyi) [inline src/modules/user/_test/usersGetAllMam.spec.js:42]: you can add one more expect for response to be an array

File: src/modules/user/_test/usersGetAllMam.spec.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    });
+
+    describe('USERS GET ALL - NEGATIVE', () => {
+
+        before('user delete all', (done) => {
+            User.deleteMany({})
+            return done()
+        })
+
+        before('user create', (done) => {
```
Comment (rustamkutovyi) [inline src/modules/user/_test/usersGetAllMam.spec.js:56]: delete before hook for user create, as you are not using it

File: src/modules/user/_test/userGetByIdMam.spec.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                query: userGetByIdQ,
+                variables: userGet
+            }
+
+            requestGql(postData)
+                .expect(200)
+                .end((err, res) => {
+                    if(err) return done(err);
+                    const respData = res.body;
+                    console.log('GET BY EMPTY ID RESP ===', respData)
```
Comment (rustamkutovyi) [inline src/modules/user/_test/userGetByIdMam.spec.js:153]: delete console

**Review by rustamkutovyi — CHANGES_REQUESTED** [review]

File: src/modules/user/_test/userUpdateByIdMam.spec.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    describe('USER UPDATE BY ID - NEGATIVE', () => {
+        let userId = null;
+
+        before('user delete all', (done) => {
+            User.deleteMany({})
+            return done()
+        })
+
+        before('user create', (done) => {
```
[the remark this answers is not in the document]
Comment (itmelia) [reply src/modules/user/_test/userUpdateByIdMam.spec.js:76]: Good point! Thanks.

File: src/modules/user/_test/usersGetAllMam.spec.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+describe('USERS GET ALL', () => {
+
+    describe('USERS GET ALL - POSITIVE', () => {
+
+        before('user delete all', (done) => {
+            User.deleteMany({})
+            return done()
+        })
+
+        before('user create', (done) => {
```
[the remark this answers is not in the document]
Comment (itmelia) [reply src/modules/user/_test/usersGetAllMam.spec.js:16]: I must leave the user create hook as the database should be empty when testing. To get users, I have to create them first.

**Review by rustamkutovyi — APPROVED** [review]

**Review by rustamkutovyi — APPROVED** [review]

**Review by rustamkutovyi — APPROVED** [review]

**Review by rustamkutovyi — APPROVED** [review]

**Pull request merged**